### DISCUSSION TITLE
変換候補に実在しない語が複数表示され、目的の候補にたどり着くのを妨げている

### DISCUSSION THREAD

**Discussion by ghost** [google/mozc#1339] [General]
シークレットモードを有効にした状態で`きねん`と入力し、スペースバーを押すと、下記の候補が現れます。
```
記念
祈念
紀年
祈年
きねん[ひらがな]
喜念
気ねん
木ねん
来ねん
杵ん
暉ねん
紀ねん
輝ねん
季ねん
（以下略）
```
「記念」〜「喜念」については異論はないのですが、「気ねん」〜「季ねん」については実在しない語と思われます。Googleでダブルクオーテーションで囲んで検索してみましたが、「気ねん」は全く見当たらず、「木ねん」は「木ねん堂」という屋号の商店こそ存在するものの、木ねんと略称する用例を見つけられませんでした。

これらの候補は、目的の候補にたどり着くのを妨げている上、誤変換を誘発しかねません。候補から消せないものでしょうか。

なお、このような不必要と思われる候補が表示されるケースは「きねん」以外にも多数存在し、ユーザー側で対処するのは極めて困難かつ場当たり的で根本的に解決できないため、ユーザー辞書の抑制単語の使用は考えておりません。

**Comment by ghost** [comment]
>不必要と思われる候補が表示されるケースは「きねん」以外にも多数存在

具体例として、「ししゅう」と入力すると「し週」「し衆」が候補に現れます。辞書の問題ではなく、変換エンジンの問題と思われます。

**Comment by ghost** [comment]
これに似た問題はサ変動詞でも発生します。
`ほんじする`と入力すると、以下のような結果が現れます。
```
翻字する
本持する
本寺する
本字する
本地する
（以下略）
```
「翻字」はサ変動詞になり得る名詞ですので、「翻字する」という候補は妥当です。
一方、本寺・本字・本地については、サ変動詞として用いられることはないと思います。本持は実在性が疑わしい語です（本持という表記の姓は実在するようですが、読みは「もともち」「ほんもち」の模様です）。

**Comment by ghost** [comment]
`からだに`でスペースキーを押すと、`過ラダに`・`可ラダに`という不要と思わしき候補が現れます。
`からだ`ではこれらの候補は現れません。

**Comment by ghost** [comment]
`かいかしょこ`でタブキーを押すと、
```
開花書庫
階下書庫
開花ショコ
開化書庫
回か書庫
回かしょこ
カイカ書庫
開架書庫
```
と表示され、肝心の開架書庫が8番目にまで下げられています。

**Comment by ghost** [comment]
`じってんぽ`と入力してスペースキーを押すと、`ジッ` `店舗`と2つに分けて変換されます。
前半の`ジッ`の変換候補は下記の通りです。
```
ジッ
十
じっ
字っ
地っ
壱拾
拾
10
１０
一〇
Ⅹ
ⅹ
⑩
字ッ
児っ
地ッ
路っ
辞っ
ぢっ
持っ
痔っ
事っ
児ッ
実
```
結果として、変換したい候補の「実」がかなり下にまで押し下げられています。字っ、地っ、字ッ〜児ッを表示しないようにできないものでしょうか。

**Comment by ghost** [comment]
`たずねる`でタブキーを押すと、以下の候補が現れます。
```
尋ねる
訪ねる
訊ねる
たずねる
タズネル
田鶴ネル
たづ寝る
たづネル
田津ネル
```
「田鶴ネル」以下については用例が確認できず、削除が望ましいと考えています。

**Comment by ghost** [comment]
`やまがたありとも`でタブキーを押すと、「山縣有朋」よりも上位に「山形アリとも」が表示されます。

**Comment by ghost** [comment]
`こうじょ`でスペースキーを押すと、下記の変換候補が現れます。
```
控除
皇女
公女
公助
公序
孝女
興除
高女
高除
好女
項所
工事ょ
校所
麹ょ
講じょ
こうじょ
小路ょ
行所
＜後略＞
```

「扣除」という言葉への変換を望んでスペースキーを押した際、「工事ょ」「麹ょ」「講じょ」「小路ょ」といった実在しない語が変換候補に現れることにより、結果として「扣除」が候補に現れないということに気付きにくくなります。
また論点が異なりますが、「高除」「項所」「行所」については用例が確認できず、語の実在性に疑問があります。

**Comment by ghost** [comment]
`きゅうじたい`で変換すると、
```
旧自体
旧事態
急自体
急事態
旧字体
(以下略)
```
と候補が表示され、目的の語である「旧字体」を5番目にまで押し下げています。

**Comment by ghost** [comment]
`どうしん`で変換すると、不要と思われる候補「`同士ん`」「`同しん`」「`動詞ん`」などが表示されます。

**Comment by yukawa** [comment]
This is also the same as #1335 (and #595). So I'm going to convert this into discussion as well.

> 辞書の問題ではなく、変換エンジンの問題と思われます。

That's true that just tweaking dictionary data in Mozc wouldn't solve above problems. If you want to discuss more on this topic, I guess starting from understanding how Mozc is implemented might be the fastest and most effective way in practice.

 - [Google 日本語入力 TechTalk 2010 Part 1 time=50:11](https://youtu.be/muJMXtuFxhs?t=3066)
 - [日本語入力を支える技術](https://gihyo.jp/book/2012/978-4-7741-4993-6)
